src/components/Column/index.d.ts

```diff
@@ -1,13 +1,13 @@
 import { ReactNode, ComponentType } from 'react';
 
-export interface ColumnProps {
+export interface ColumnProps<P> {
     header?: ReactNode;
-    component?: ComponentType;
+    component?: ComponentType<P>;
     field?: string;
     sortable?: boolean;
     width?: number | string;
     defaultWidth?: number | string;
     type?: 'text' | 'action';
 }
 
-export default function(props: ColumnProps): JSX.Element | null;
+export default function<P>(props: ColumnProps<P>): JSX.Element | null;
```

Comment:
we need to let the user pass extra props to Column
```suggestion
export default function<P>(props: ColumnProps<P>  & P): JSX.Element | null;
```